docs/rules/order.md

```diff
@@ -592,6 +593,135 @@ This happens because [type-only imports][6] are considered part of one global
 
 The same example will pass.
 
+### `newlines-between-types`
+
+Valid values: `"ignore" | "always" | "always-and-inside-groups" | "never"` \
+Default: the value of [`newlines-between`][20]
+
+> \[!NOTE]
+>
+> This setting is only meaningful when [`sortTypesGroup`][7] is enabled.
```

Comment:
should we use the schema to ensure that this option can't be set to `true` unless `sortTypesGroup` is also set to true?

Reply:
Sounds like a good idea to me. I was not aware that was a thing :)

Reply:
it often requires a bit of gymnastics, but it's usually worth it for the DX.

Reply:
@Xunnamius is this change something you're planning to make?

Reply:
@ljharb Yep, got a bit of free time just now